src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/judgeapproval/JudgeApprovalResolver.java

```diff
@@ -100,18 +93,22 @@ void processApprovableCollection(List<? extends Approvable> approvables, CaseDoc
      * @param userAuthorisation the user authorization string to get the judge's full name
      */
     void handleApprovable(Approvable approvable, JudgeApproval judgeApproval, String userAuthorisation) {
-        approvable.setApprovalJudge(idamService.getIdamFullName(userAuthorisation));
-        if (isJudgeApproved(judgeApproval)) {
-            if (judgeApproval.getJudgeDecision() == JUDGE_NEEDS_TO_MAKE_CHANGES) {
-                approvable.replaceDocument(judgeApproval.getAmendedDocument());
+        if (isJudgeApproved(judgeApproval) || isJudgeRefused(judgeApproval)) {
```

Comment:
If either isJudgeApproved or isJudgeRefused have to do a lot of processing to get the result then you would duplicate all that effort as you will always call the same function twice.

Reply:
Resolved - [7152a62](https://github.com/hmcts/finrem-case-orchestration-service/pull/2051/commits/7152a62aa0b72a0c1d39106731e336b984a21fc1)